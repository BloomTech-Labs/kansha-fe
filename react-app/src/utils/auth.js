import auth0 from 'auth0-js';
import jwtDecode from 'jwt-decode';

export default class Auth {
	auth0 = new auth0.WebAuth({
		domain: process.env.REACT_APP_AUTH_DOMAIN,
		clientID: process.env.REACT_APP_CLIENT_ID,
		redirectUri: `${process.env.REACT_APP_REDIRECT_URI_BASE}/auth`,
		audience: `https://${process.env.REACT_APP_AUTH_DOMAIN}/userinfo`,
		responseType: 'token id_token',
		scope: 'openid profile',
	});

	constructor() {
		this.login = this.login.bind(this);
	}

	login() {
		this.auth0.authorize();
	}

	handleAuthentication(cb) {
		console.log('foo')
		this.auth0.parseHash((err, authResults) => {
			if (authResults && authResults.accessToken && authResults.idToken) {
				let expiresAt = JSON.stringify(
					authResults.expiresIn * 1000 + new Date().getTime(),
				);
				localStorage.setItem('access_token', authResults.accessToken);
				localStorage.setItem('id_token', authResults.idToken);
				localStorage.setItem('expires_at', expiresAt);
				window.location.hash = '';
				console.log('hello world')
				cb()
			} else if (err) {
				console.log(err)
				cb(err)
			}
		});
	}

	isAuthenticated() {
		let expiresAt = JSON.parse(localStorage.getItem('expires_at'));
		return new Date().getTime() < expiresAt;
	}

	getProfile() {
		if (localStorage.getItem('id_token')) {
			return jwtDecode(localStorage.getItem('id_token'));
		} else {
			return {};
		}
	}
}
