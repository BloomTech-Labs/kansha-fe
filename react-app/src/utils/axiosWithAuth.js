import axios from 'axios';

export const axiosWithAuth = () => {
  const token = localStorage.getItem('id_token');
  return axios.create({
    headers: {
      Authorization: token
    },
    baseURL: process.env.REACT_APP_BASE_URL
  });
};