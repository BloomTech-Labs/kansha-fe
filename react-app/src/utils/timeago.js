/* Generates realative time strings */

export const timeAgo = timestamp => {
    const currentTime = new Date(Date.now()).getTime();
    const eventTime = new Date(timestamp).getTime()

	const elapsedTime = currentTime - eventTime;

	const minute = 60 * 1000;
	const hour = minute * 60;
	const day = hour * 24;
	const week = day * 7;
	const month = week * 4;
	const year = day * 365;

	if (elapsedTime < minute) {
		return 'now';
	} else if (elapsedTime < hour) {
		let time = Math.round(elapsedTime / minute);
		return `${time} minute${time > 1 ? 's' : ''} ago`;
	} else if (elapsedTime < day) {
		let time = Math.round(elapsedTime / hour);
		return `${time} hour${time > 1 ? 's' : ''} ago`;
	} else if (elapsedTime < week) {
		let time = Math.round(elapsedTime / day);
		return `${time} day${time > 1 ? 's' : ''} ago`;
	} else if (elapsedTime < month) {
		let time = Math.round(elapsedTime / week);
		return `${time} week${time > 1 ? 's' : ''} ago`;
	} else if (elapsedTime < year) {
		let time = Math.round(elapsedTime / month);
		return `${time} month${time > 1 ? 's' : ''} ago`;
	} else {
		let time = Math.round(elapsedTime / year);
		return `${time} year${time > 1 ? 's' : ''} ago`;
	}
};
