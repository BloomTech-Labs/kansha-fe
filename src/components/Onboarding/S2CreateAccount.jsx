import React from 'react';
import { useHistory } from 'react-router-dom';
import { axiosWithAuth } from '../../utils/axiosWithAuth';

// post request to /users endpoint to create new user

function S2CreateAccount({ user, setUser, handleUser }) {
	let history = useHistory();

	const handleSubmit = e => {
		e.preventDefault();
		axiosWithAuth()
			.post('/users', user)
			.then(res => {
				console.log(res, `hello`);
				setUser({ ...user, id: res.data[0] });
			})
			.catch(err => console.log(err));
		history.push('/onboarding/step-3');
	};

	return (
		<div>
			<h1>Create Account</h1>
			<form className="create-account-form">
				<div className="name-container">
					<input
						className="formname"
						placeholder="First Name"
						name="first_name"
						value={user.first_name}
						onChange={handleUser}></input>
					<input
						className="formname"
						placeholder="Last Name"
						name="last_name"
						value={user.last_name}
						onChange={handleUser}></input>
				</div>

				<input
					className="jobtitle-input"
					placeholder="Job Title"
					name="job_title"
					value={user.job_title}
					onChange={handleUser}></input>
				<input
					placeholder="Email"
					name="email"
					value={user.email}
					onChange={handleUser}></input>
				<input
					placeholder="Organization Name"
					name="org_name"
					value={user.org_name}
					onChange={handleUser}></input>
			</form>
			<button type="submit" onClick={handleSubmit}>
				Next
			</button>
		</div>
	);
}

export default S2CreateAccount;
