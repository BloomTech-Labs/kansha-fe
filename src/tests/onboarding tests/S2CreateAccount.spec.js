import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import S2CreateAccount from '../../components/Onboarding/S2CreateAccount';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { renderWithRouter } from '../utils';

let S2Component = (
	<BrowserRouter>
		<S2CreateAccount
			user={{
				first_name: 'Vanessa',
				last_name: 'Staneck',
				job_title: 'manager',
				org_name: 'kansha',
				org_name: 'less than 20',
				industry: 'accounting',
			}}
		/>
	</BrowserRouter>
);

afterEach(cleanup);

describe.skip('S2', () => {
	it('matches snapshot', () => {
		const tree = renderer.create(S2Component);
		expect(tree.toJSON()).toMatchSnapshot();
	});

	it('renders without crashing', () => {
		render(S2Component);
	});

	it('should call the next button when clicked', () => {
		const { getByText, history } = renderWithRouter(
			<S2CreateAccount
				user={{
					first_name: 'Vanessa',
					last_name: 'Staneck',
					job_title: 'manager',
					org_name: 'kansha',
					org_name: 'less than 20',
					industry: 'accounting',
				}}
			/>,
		);

		const nextButton = getByText(/Next/i);
		fireEvent.click(nextButton);
		expect(history.location.pathname).toBe('/onboarding/step-4');
	});

	it('it displays org size question', () => {
		const { getByText, getAllByText } = render(S2Component);
		getByText(/How big is your organization?/i);
		getByText(/less than 20/i);
		getByText(/21 to 100/i);
		getByText(/over 100/i);
	});
});
