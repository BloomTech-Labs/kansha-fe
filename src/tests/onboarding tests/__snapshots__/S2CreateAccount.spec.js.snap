// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`S2 matches snapshot 1`] = `
<div>
  <h1>
    Let's get set up!
  </h1>
  <form
    className="create-account-form"
  >
    <div
      className="name-container"
    >
      <input
        className="formname"
        defaultValue="Joss"
        name="first_name"
        placeholder="First Name"
      />
      <input
        className="formname"
        defaultValue="Staneck"
        name="last_name"
        placeholder="Last Name"
      />
    </div>
    <input
      className="formname"
      defaultValue="manager"
      name="job_title"
      placeholder="Job Title"
    />
    <h6>
      What's your organization name?
    </h6>
    <input
      className="orgname-input"
      defaultValue="less than 20"
      name="org_name"
      placeholder="Organization Name"
    />
    <h6>
      How big is your organization?
    </h6>
    <div
      className="org-size"
    >
      <div
        className="radio-div"
      >
        <input
          defaultValue="  less than 20"
          id="lessthan20"
          name="company_size"
          type="radio"
        />
      </div>
      <label
        htmlFor="lessthan20"
      >
        Less than 20
      </label>
      <div
        className="radio-div"
      >
        <input
          defaultValue="  21 to 100"
          id="21100"
          name="company_size"
          type="radio"
        />
        <label
          htmlFor="21100"
        >
          21 - 100
        </label>
      </div>
      <div
        className="radio-div"
      >
        <input
          defaultValue="  over 100"
          id="over100"
          name="company_size"
          type="radio"
        />
        <label
          htmlFor="over100"
        >
          Over 100
        </label>
      </div>
    </div>
    <div
      className="dropdown-container"
    >
      <label
        className="industry-label"
        htmlFor="industry"
      >
        Select your industry:
      </label>
      <div
        className="custom-select"
      >
        <select>
          <option
            value="Accounting"
          >
            Accounting
          </option>
          <option
            value="Advertising/PR"
          >
            Advertising/PR
          </option>
          <option
            value="Aerospace"
          >
            Aerospace
          </option>
          <option
            value="Argiculture"
          >
            Agriculture
          </option>
          <option
            value="Architecture"
          >
            Architecture
          </option>
          <option
            value="Airlines"
          >
            Airlines
          </option>
          <option
            value="Automotive"
          >
            Automotive
          </option>
          <option
            value="Banking/Finance"
          >
            Banking/Finance
          </option>
          <option
            value="Business (general)"
          >
            Business (general)
          </option>
          <option
            value="Communications"
          >
            Communications
          </option>
          <option
            value="Education"
          >
            Education
          </option>
          <option
            value="Entertainment"
          >
            Entertainment
          </option>
          <option
            value="Hospitality"
          >
            Hospitality
          </option>
          <option
            value="IT/Computers/Technology"
          >
            IT/Computers/Technology
          </option>
          <option
            value="Legal"
          >
            Legal
          </option>
          <option
            value="Medical/Health Services"
          >
            Medical/Health Services
          </option>
        </select>
        <div
          className="select-selected"
          id="industry"
          onClick={[Function]}
        >
          <div
            className="placeholder-div"
          >
            e.g. Accounting
          </div>
        </div>
        <div
          className="select-items select-hide"
        >
          <div
            data-value="Accounting"
            onClick={[Function]}
          >
            Accounting
          </div>
          <div
            data-value="Advertising/PR"
            onClick={[Function]}
          >
            Advertising/PR
          </div>
          <div
            data-value="Aerospace"
            onClick={[Function]}
          >
            Aerospace
          </div>
          <div
            data-value="Argiculture"
            onClick={[Function]}
          >
            Agriculture
          </div>
          <div
            data-value="Architecture"
            onClick={[Function]}
          >
            Architecture
          </div>
          <div
            data-value="Airlines"
            onClick={[Function]}
          >
            Airlines
          </div>
          <div
            data-value="Automotive"
            onClick={[Function]}
          >
            Automotive
          </div>
          <div
            data-value="Banking/Finance"
            onClick={[Function]}
          >
            Banking/Finance
          </div>
          <div
            data-value="Business (general)"
            onClick={[Function]}
          >
            Business (general)
          </div>
          <div
            data-value="Communications"
            onClick={[Function]}
          >
            Communications
          </div>
          <div
            data-value="Education"
            onClick={[Function]}
          >
            Education
          </div>
          <div
            data-value="Entertainment"
            onClick={[Function]}
          >
            Entertainment
          </div>
          <div
            data-value="Hospitality"
            onClick={[Function]}
          >
            Hospitality
          </div>
          <div
            data-value="IT/Computers/Technology"
            onClick={[Function]}
          >
            IT/Computers/Technology
          </div>
          <div
            data-value="Legal"
            onClick={[Function]}
          >
            Legal
          </div>
          <div
            data-value="Medical/Health Services"
            onClick={[Function]}
          >
            Medical/Health Services
          </div>
        </div>
      </div>
    </div>
  </form>
  <button
    disabled={true}
    onClick={[Function]}
    type="submit"
  >
    Next
  </button>
</div>
`;
